### PR TITLE
HW7: add mongoDB

### PR DESCRIPTION
**Pull request by YuliaBarbashova**

### CODE REVIEW

File: HW7/repositories/cart.repository.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+export const getCartByUserId = (id: string) => {
+  return CartModel.getCart(id);
+};
+
+export const createNewCart = (id: string): Promise<CartEntity> => {
+  return CartModel.createCart(id);
+};
+
+export const updateCardByUserId = (
```
Comment (AlexeyBarkow) [inline HW7/repositories/cart.repository.ts]: a typo: uptadeCart

File: HW7/controllers/cart.controller.ts [hunk 2]
```diff
@@ -0,0 +1,129 @@
+import { Router } from "express";
+
+import { CartService } from "../services/cart.service.ts";
+import { authMiddleware } from "../middlewares/authMiddleware.ts";
+import { CartEntity, OrderEntity } from "../types/index.ts";
+import { validateBody, cartSchema } from "../validators/index.ts";
+
+const CartController = Router();
+
+CartController.get("/", authMiddleware, async (req, res, next) => {
```
Comment (AlexeyBarkow) [inline HW7/controllers/cart.controller.ts:10]: Cart deletion works partially: when I remove cart or submit checkout, I still receive cart with isDeleted: true

File: HW7/models/carts.model.ts [hunk 3]
```diff
@@ -0,0 +1,80 @@
+import Cart, { ICart } from "./schemas/cart.schema.ts";
+
+import { CartEntity, CartItemEntity, ProductEntity } from "../types/index.ts";
+
+const getCart = (id: string): Promise<ICart | null> => {
+  return new Promise((resolve, reject) => {
```
Comment (AlexeyBarkow) [inline HW7/models/carts.model.ts]: This looks very strange: you return promise which resolves synchronously with promise (Cart.findOne returns promise). Even though this works, I'd recommend to rewrite this:
```
const getCart = async (id: string): Promise<ICart | null> => {
  const cart = await Cart.findOne({ userId: id });

  return cart;
};
```

**Review by AlexeyBarkow — CHANGES_REQUESTED** [review]

File: HW7/repositories/cart.repository.ts (hunk 1 above)
Comment (YuliaBarbashova) [reply HW7/repositories/cart.repository.ts:12]: ???

Comment (YuliaBarbashova) [reply HW7/repositories/cart.repository.ts]: Updated. You have a typo too ;)

File: HW7/controllers/cart.controller.ts (hunk 2 above)
Comment (YuliaBarbashova) [reply HW7/controllers/cart.controller.ts:10]: Updated

File: HW7/models/carts.model.ts (hunk 3 above)
Comment (YuliaBarbashova) [reply HW7/models/carts.model.ts]: Updated.
Thanks for review)

**Pull request merged**